src/PYTHON/python_impl.cpp

```diff
@@ -416,10 +440,22 @@ int PythonImpl::find(const char *name)
   return -1;
 }
 
-/* ------------------------------------------------------------------ */
-
-int PythonImpl::variable_match(const char *name, const char *varname, int numeric)
+/* ---------------------------------------------------------------------
+   called by Variable class when a python-style variable is evaluated
+     this will call invoke_function() in this class
+   either via Variable::retrieve() or Variable::equalstyle
+     retrieve calls with numeric = 0, equalstyle with numeric = 1
+   ensure name matches a Python function
+   ensure the Python function produces an output
+   ensure the Python function outputs to the matching python-style variable
+   ensure a string is returned only if retrieve() is the caller
+--------------------------------------------------------------------- */
+
+int PythonImpl::function_match(const char *name, const char *varname, int numeric)
 {
+  // NOTE to Richard - any reason not to put error messages here instead of in Variable class ?
+  //   error messages appear 3x in Variable
+
```

Comment:
@sjplimp The main reason is to not have PytonImpl be a Pointers subclass. We can avoid that by passing `error` or raising an exception instead.

Reply:
moved error handling to PythonImpl class and out of Variable class.